tests/integration/operator/verify_test.go

```diff
@@ -57,12 +64,49 @@ func TestPostInstallControlPlaneVerification(t *testing.T) {
 			}
 
 			t.Cleanup(func() {
-				uninstallCmd := []string{
-					"experimental",
-					"uninstall",
-					"--purge", "-y",
+				if err = purgeIstioResources(istioCtl); err != nil {
+					logClusterState(cs)
+					t.Fatalf("error during cleanup: %v", err)
 				}
-				istioCtl.InvokeOrFail(t, uninstallCmd)
 			})
 		})
 }
+
+func cleanupCluster(t *testing.T, cs resource.Cluster, istioCtl istioctl.Instance) {
+	cleanupInClusterCRs(t, cs)
+	if err := purgeIstioResources(istioCtl); err != nil {
+		scopes.Framework.Warnf("Failed to purge istio resources: %v", err)
+	}
+	nsList := []string{IstioNamespace, OperatorNamespace}
+	for _, ns := range nsList {
+		if err := cs.CoreV1().Namespaces().Delete(context.TODO(), ns, kube2.DeleteOptionsForeground()); err != nil {
```

Comment:
Delete namespace is super slow, and you need to wait for it - we are not here, which is why the tests fail. 

Why do we need to delete the namespace? Doesn't that pretty much mean the operator has a real bug where it fails to cleanup properly and we are masking that by deleting the namespace? 

We should be testing what our users do.

Reply:
Also doesn't deleting the namespace actually not work since the finalizer on the IOP will get stuck?

Reply:
Thanks. Let me resolve that.

Reply:
You're right. Test is working fine after removing it. No timeouts. In this PR, I added Istio uninstallation as it could mess up verification.